client/blocks/edit-gravatar/index.jsx

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import classnames from 'classnames';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FilePicker from 'components/file-picker';
+import FormLabel from 'components/forms/form-label';
+import {
+	getCurrentUser,
+	isCurrentUserUploadingGravatar
+} from 'state/current-user/selectors';
+import Gravatar from 'components/gravatar';
+import { isOffline } from 'state/application/selectors';
+import { localize } from 'i18n-calypso';
+import Spinner from 'components/spinner';
+import * as OAuthToken from 'lib/oauth-token';
+import { uploadGravatar } from 'state/current-user/actions';
+
+export class EditGravatar extends Component {
+	constructor() {
+		super();
+		this.handleOnPick = this.handleOnPick.bind( this );
+	}
+
+	static propTypes = {
+		isOffline: PropTypes.bool,
+		translate: PropTypes.func,
+		user: PropTypes.object,
+		dispatch: PropTypes.func,
+		isUploading: PropTypes.bool,
+	};
+
+	handleOnPick( files ) {
+		const { dispatch, user } = this.props;
+		console.log( 'you picked',  JSON.stringify( files[0].name ) );
+
+		// check for bearerToken from desktop app
+		let bearerToken = OAuthToken.getToken();
+
+		if ( ! bearerToken ) {
+			bearerToken = localStorage.getItem( 'bearerToken' );
+		}
+
+		// send gravatar request
+		if ( bearerToken ) {
+			console.log( 'Got the bearerToken, sending request' );
+			dispatch( uploadGravatar( files[ 0 ], bearerToken, user.email ) );
```

Comment:
Ah, yes! Thank you! I was trying to get this to work last night, but forgot to pull `uploadGravatar` from the props first, so it was not co-operating :P.

I'm using `this.props.uploadGravatar` properly now, and I cleaned it all up in b2a8380 :D